graphix/optimization.py

```diff
@@ -418,6 +607,36 @@ def _incorporate_pauli_results_in_domain(
     return odd_outcome == 1, new_domain
 
 
+def _update_corrections(
+    node: Node, domain: AbstractSet[Node], correction: dict[Node, set[Node]]
+) -> dict[Node, set[Node]]:
+    """Update the correction mapping by adding a node to all entries in a domain.
+
+    Parameters
+    ----------
+    node : Node
+        The node to add as a correction.
+    domain : AbstractSet[Node]
+        A set of measured nodes whose corresponding correction sets should be updated.
+    correction : dict[Node, set[Node]]
+        A mapping from measured nodes to sets of nodes on which corrections are applied. This
+        dictionary is modified in place.
+
+    Returns
+    -------
+    dict[Node, set[Node]]
+        The updated correction dictionary with `node` added to the correction
+        sets of all nodes in `domain`.
+
+    Notes
+    -----
+    This function is used to extract the correction function from :math:`X`, :math:`Z` and :math:`M` commands when constructing a flow.
+    """
+    for measured_node in domain:
+        correction[measured_node].add(node)
```

Comment:
If `correction` is just a `dict`:

```suggestion
        correction.setdefault(measured_node, set()).add(node)
```